### ISSUE TITLE
user --help should override built-in

### ISSUE THREAD

**Issue by bradcray** [chapel-lang/chapel#4973]
### Summary of Problem

It seems to me that defining a config named 'help' should take precedence in argument parsing over the built-in '--help' flag that all Chapel executables support.  E.g., for the example below, I'd expect printUsage() to be invoked when running with --help, but instead the compiler-provided help message is printed.

### Steps to Reproduce

**Source Code:**

This is taken from `test/execflags/bradc/configHelp.chpl`:

```chapel
config const help = false;
if false then printUsage();

proc printUsage() {
  writeln("This is my custom usage message!");
  exit(0);
}
```

**Compile command:**

`chpl configHelp.chpl`

**Execution command:**

`./a.out --help`

### Configuration Information

- Output of `chpl --version`: 1.14.0
- Output of `$CHPL_HOME/util/printchplenv --anonymize`: N/A
- Back-end compiler and version, e.g. `gcc --version` or `clang --version`: N/A
- (For Cray systems only) Output of `module list`: N/A

**Comment by mppf** [comment]
It seems odd that the example has an 'if false' - is that and error in the example? I suppose that either way you see the compiler-generated help.
Anyway a good example of user-overriding confit help should also call Help.printUsage() (from the Help standard module).

**Comment by bradcray** [comment]
Oops, I think what happened was that I changed the test to 'true' to verify the .good file and then switched it to 'false' instead of 'help' when setting it back again.

I'm not sure I agree that good examples of providing custom help should necessarily call Help.printUsage()...  I.e., maybe that's not what they would want the help message to look like.

**Comment by arezaii** [comment]
With the addition of the `ArgumentParser` module in v1.25, do we still want to try to support this behavior?

**Comment by bradcray** [comment]
@arezaii : I'm not seeing how the addition of `ArgumentParser` changes the root issue here (that defining a `config` whose name happens to match a built-in flag silently loses to that flag).  Does it?

**Comment by arezaii** [comment]
No, you're right, it doesn't address this issue where a `config` shadows a built-in. I was imagining the user could handle the `--help` flag with the `ArgumentParser` instead of a `config`. 

Do you envision a fix for this to handle just the `--help` flag, or all/any Chapel defined flags, or maybe some other subset of flags?

**Comment by bradcray** [comment]
I don't have a specific fix in mind, but I think the current behavior could potentially be pretty confusing to a user.  I think with this specific case, it's easy to say "Well, the user should just use one of the other help capabilities that we support to get the help message they want", but I think the larger issue is that they may be using a config named `help` to do something other than print out help / may not know about those features / may run into this with other similar flags.

So a very simple short-term fix would be to generate a warning for such configs indicating that they're shadowed by other built-in flags and that the user will have to use `-s` or a fully-qualified `--moduleName.help=` flag to set them (or should rename the variable to avoid the shadowing).

Longer-term, this relates to the discussion we've been having about the interaction between built-in flags (whether for built-in configs or other features) and arguments sent to main().
